### PR TITLE
Add missing tests, add validation for unique jobs name

### PR DESCRIPTION
**Pull request by jasiu001**
Related to #487

### CODE REVIEW

File: development/README.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ The `development` folder has the following structure:
   ├── update-config.sh         # This script updates the new configuration of the "config.yaml" file on a cluster.
   ├── update-jobs.sh           # This script updates the new configuration of the jobs on a cluster.
   ├── update-plugins.sh        # This script updates the new configuration of the "plugins.yaml" file on a cluster.
-  ├── validate-config.sh       # This script runs the "Checker" application.
+  ├── validate-config.sh       # This script runs the "Checker" application and test which checks uniqueness of jobs name.
```
Comment (klaudiagrz) [inline development/README.md]:
```suggestion
  ├── validate-config.sh       # This script runs the "Checker" application and checks the uniqueness of jobs names.
```

File: development/checker/unique_jobs_name/main.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
+package main
```
[the remark this answers is not in the document]
Comment (jasiu001) [reply development/checker/unique_jobs_name/main.go]: I changed to `unique-job-names` like other files/directories convention with multiple name in this repository

File: development/checker/unique-jobs-name/main.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+	"os"
+)
+
+type options struct {
+	prowConfig   string
+	jobConfigDir string
+}
+
+type report struct {
+	active   bool
```
Comment (aszecowka) [inline development/checker/unique-jobs-name/main.go]: consider renaming `active` to `foundDuplicates`

File: development/checker/unique-jobs-name/main.go [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+			rep.messages = append(rep.messages, fmt.Sprintf("Prow job %q has %d instances", name, val))
+		}
+	}
+
+	if rep.active {
+		logrus.Fatalf("Config jobs are not unique:\n%s", rep.toString())
+	}
+}
+
+func addPreSubmitJobsName(jobs map[string]int, c map[string][]config.Presubmit) {
```
[the remark this answers is not in the document]
Comment (jasiu001) [reply development/checker/unique-jobs-name/main.go]: 👍

File: development/checker/unique-jobs-name/main.go (hunk 3 above)
Comment (jasiu001) [reply development/checker/unique-jobs-name/main.go]: 👍

**Pull request merged**